src/index.tsx

```diff
@@ -1,17 +1,30 @@
-import { RefObject, useEffect, useState } from 'react';
+import { RefObject, useEffect, useRef, useState } from 'react';
 
-interface Options {
+export interface Options {
   accept?: string;
   multiple?: boolean;
 }
 
-interface Props {
+export interface Args {
   ref: RefObject<HTMLInputElement>;
   options?: Options | undefined;
 }
 
-function useInputFile({ ref, options }: Props) {
+export interface Callback {
+  (event: Event): void;
+}
+
+export type Files = {
+  files: FileList | null;
+};
+
+function useInputFile({ ref, options }: Args, callback?: Callback): Files {
   const [files, setFiles] = useState<FileList | null>(null);
+  const customerHandler = useRef<(event: Event) => void>();
+
+  useEffect(() => {
+    customerHandler.current = callback;
```

Comment:
so you want the ref to be updated after each mount?

Reply:
I think it's not a good design for here.

Save callback in ref without `useEffect`:

```ts
const customerHandler = useRef<Callback | undefined>(undefined);
customerHandler.current = callback;
```

Reply:
👍